API integration of Signin, Signup and User Data

shafat730 opened the pull request:
## Description
<!-- Link your ticket using #{ISSUE_NUMBER}. And add a clear and concise description of what this PR does. -->
closes #615 
Integrate Sign In, Sign Up and User Data functionality in a React frontend using custom Axios-based API calls and written test cases to ensure the components behave correctly.

CSS update:
Moved the error-related CSS classes (.error-container, .input-error, .has-error) to the global CSS file(index.css).
Updated components to use regular class names instead of CSS module styles (imported index.css in App.jsx).
Changes done in SignIn, SignUp and CustomInput component. 

For SignIn:

- Handled the request through the useApi custom hook for better reusability and state handling.
- On successful SignIn, the header items is updated to reflect the user's login status.
- Integrated [MSW (Mock Service Worker)] for testing API behavior without hitting real endpoints.
- Created mock handlers for SignIn .

For SignUp:

- Integrates Signup api. Covers the functionalities and test cases both.
- The signup route is implemented , the user can signup and after successful signup user will be toggled to signin component.
- Code change also includes changes in .env.example file:

.env.example(ui) : BASE_URL is changed to VITE_BASE_URL , reason being url was not accessible.
.env.example(app): Included ADMINSEMAILS as it was created issue and was not includes by me in Add functionality to signup Admins. #562 .

For GetUserData API:

- Integrated getuserdata api in Usage.tsx & UserInfo.tsx Component.
- added disabled prop in CustomInput component.
- Added test-cases for Usage.tsx & UserInfo.tsx component.

Extras:

- Header.tsx & MobileHeaderMenu.tsx (ui): conditionally render /admin based on user-role.
- auth.js (app): added userRole along with 200 status-code in signin function so we can redirect user after sign-in based on their role.
- Catalog.jsx(ui): added name attribute (coz getting warning).
- ApiKeyForm.jsx(ui): added value attribute (coz getting warning).


## What type of PR is this? (Check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📄 Documentation Update
- [ ] 👨‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🛠️ CI/CD

## Screenshots (if applicable)
<!-- Add screenshots to help explain your changes. -->
![image](https://github.com/user-attachments/assets/ae2f7264-6181-4e09-b517-f5dcd0dbc2e1)
![image](https://github.com/user-attachments/assets/93da9ab9-3f3c-4651-89f9-9d0d5c012ee4)


https://github.com/user-attachments/assets/34170bf6-d936-437a-b3a2-8b4baed815ae





## Checklist
- [ ] I have performed a self-review of my code  
- [ ] I have commented my code, particularly in hard-to-understand areas  
- [x] I have added tests that prove my fix is effective or that my feature works  
- [x] New and existing unit tests pass locally with my changes

shafat730 committed:
Refactor authentication logic and enhance UI components with context integration

shafat730 committed:
Refactor authentication logic and enhance UI components with context integration

theboyofdream committed:
Merge branch 'stage' of https://github.com/TeamShiksha/openlogo into 615-integrating-apis

strawHat121 committed:
Refactor tests to integrate AuthContext and UserContext in Header, MobileHeaderMenu, Auth, and SignIn components

theboyofdream committed:
Merge branch '615-integrating-apis' of https://github.com/TeamShiksha/openlogo into 615-integrating-apis

shafat730 committed:
Refactor MobileHeaderMenu test to correctly render when isOpen is true; remove unused SignIn form reset test; enhance SignIn component error handling with CSS transitions

Severin84 committed:
Signup integration

Severin84 committed:
added error response feedback in signu
p component

Severin84 committed:
two packages combined

shafat730 committed:
Merge pull request #625 from Severin84/615-integrating-apis

Signup integration

shafat730 committed:
Added api integration for signup  and signup and their test cases

theboyofdream committed:
Merge branch '615-integrating-apis' of https://github.com/TeamShiksha/openlogo into 615-integrating-apis

aps08 committed:
Merge branch 'stage' into 615-integrating-apis

theboyofdream committed:
Merge branch '615-integrating-apis' of https://github.com/TeamShiksha/openlogo into 615-integrating-apis

shafat730 committed:
Refactor header navigation and update error handling in SignIn component; add mock form values for testing

theboyofdream committed:
Merge branch '615-integrating-apis' of https://github.com/TeamShiksha/openlogo into 615-integrating-apis

shafat730 committed:
Merge branch 'stage' into 615-integrating-apis

shafat730 committed:
Merge branch 'stage' of github.com:TeamShiksha/logoexecutive into 615-integrating-apis

theboyofdream committed:
integrating getuserdata API in dashboard

theboyofdream committed:
Merge branch '615-integrating-apis' of https://github.com/TeamShiksha/openlogo into 615-integrating-apis

shafat730 committed:
Merge branch '615-integrating-apis' of github.com:TeamShiksha/logoexecutive into 615-integrating-apis

shafat730 committed:
refactor: remove unused mockFormvalues and integrate it directly in SignIn tests

aps08 commented on packages/app/controllers/auth.js:
```diff
@@ -135,7 +135,7 @@ async function signinController(req, res, next) {
     });
 
-    return res.status(200).json({ statusCode: 200 });
+    return res.status(200).json({ statusCode: 200, userRole: user.role });
```
what is this for ?

theboyofdream replied:
so we can redirect user after successful sign-in to dashboard based on their role. if admin sign-in then we will redirect it to `/admin` route otherwise `/dashboard` route. 

and we are also showing `Admin` link  in Header if admin sign-in

shafat730 committed:
refactor: remove unused mockFormvalues and integrate it directly in SignIn tests

aps08 commented on packages/ui/src/mocks/handlers/signinhandler.js:
```diff
@@ -0,0 +1,7 @@
+import { http, HttpResponse } from "msw";
+
+export const signinHandler = [
+  http.post(`${import.meta.env.VITE_BASE_URL}/api/auth/signin`, async () => {
```
what is this ? Is this how we are going to use msw ? If we have to make API calls directly to the backend then what is the user of using msw?

aps08 commented on packages/ui/src/components/dashboard/userinfo/UserInfo.jsx:
```diff
@@ -1,10 +1,13 @@
+import { UserContext } from "../../../contexts/Contexts";
 
 function UserInfo() {
+  const { userData, loading } = useContext(UserContext);
```
user data contain some information like table information if we use it here then how are you going to pass the table data into table component.

aps08 commented on packages/ui/src/components/dashboard/usage/Usage.jsx:
```diff
@@ -1,22 +1,46 @@
+import { UserContext } from "../../../contexts/Contexts";
 
 function Usage() {
+  const { userData, loading } = useContext(UserContext);
```
usage is again a child component of dashboard. Why to use `useContext` in every child. Why don't we use prop drilling for this ?

aps08 commented on packages/ui/src/components/auth/Signin.jsx:
```diff
@@ -40,19 +49,35 @@ const SignIn = ({ toggleForm }) => {
       <form className={styles.form} onSubmit={handleSubmit}>
         <img src="/logo-images.png" alt="openlogo" className={styles.logo} />
         <h2 className={styles.title}>{SIGNIN.title}</h2>
+        <div
```
do we need a div just around a p `tag` here ?

aps08 commented on packages/ui/setupTest.js:
did we run prettier here ?

aps08 commented on packages/ui/src/components/auth/SignForm.module.css:
please confirm if .has-error or .input-error is already defined in the root css.

aps08 commented on packages/ui/src/components/auth/Signup.jsx:
```diff
@@ -47,6 +53,13 @@ function SignUp({ toggleForm }) {
+    const success = await makeRequest();
+    if (success) {
+      setFormValues(SIGNUP.initialValues);
+      setIsSubmit(true);
```
if we toggle instantly then user won't see the success message on the UI

aps08 commented on packages/ui/src/components/auth/Signup.jsx:
```diff
@@ -58,6 +71,11 @@ function SignUp({ toggleForm }) {
         onSubmit={handleSubmit}
       >
         <h2 className={styles.title}>{SIGNUP.title}</h2>
+        <div
```
do we need div wrapper around p

aps08 commented on packages/ui/src/components/header/MobileHeaderMenu.jsx:
```diff
@@ -21,12 +24,33 @@ const MobileHeaderMenu = ({ closeMenu, isOpen }) => {
     <div data-testid="mobile-menu" className={styles["mobile-header"]}>
       <div className={styles.navbar}>
-        {HEADER_ITEMS.map((item) => (
+        {userData && userData.role === "ADMIN" && (
```
as mentioned don't take admin into consideration for now

aps08 reviewed (changes requested):
work on the requested changes.

shafat730 committed:
Merge branch 'stage' into 615-integrating-apis

shafat730 committed:
refactor: update error handling styles and define it in root css

theboyofdream committed:
- removed unnecessary code from signin, auth
- refactored code of user-info & usage component
- added test-case for dashboard
